### PR TITLE
feat(sidecar): Add Rust sidecar

### PR DESCRIPTION
**Pull request by sunix**
### What does this PR do?
Add a side car for the rust extension

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/17069

### How to test this PR?
Run a workspace out of this devfile.yaml:
```yaml
apiVersion: 1.0.0
metadata:
  generateName: rust-
projects:
  - name: HelloWorld-Rust
    source:
      location: 'https://github.com/che-samples/helloworld-rust.git'
      type: git
      branch: master
components:
  - mountSources: true
    memoryLimit: 512Mi
    type: dockerimage
    alias: rust
    image: quay.io/eclipse/che-rust-1.39:nightly
 
  - type: chePlugin
    reference: 'https://gist.githubusercontent.com/sunix/eb86fdf6802afbfb4de059ed19f08d37/raw/41515915211723ca3c740d28263d7b8a87e0a8b6/meta.yaml'
 
commands:
  - name: build
    actions:
      - workdir: '${CHE_PROJECTS_ROOT}/HelloWorld-Rust'
        type: exec
        command: cargo build
        component: rust
  - name: run
    actions:
      - workdir: '${CHE_PROJECTS_ROOT}/HelloWorld-Rust'
        type: exec
        command: cargo run
        component: rust
```
Open the rust file.
Observe autocomplete is working.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: sidecars/rust/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM rust:1.48.0-slim
```
Comment (benoitf) [inline sidecars/rust/Dockerfile:1]:
```suggestion
FROM rust:1.48.0-slim-buster
```

File: sidecars/rust/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+ENV HOME=/home/theia
+
+RUN mkdir /projects ${HOME} && \
+    # Change permissions to let any arbitrary user
+    for f in "${HOME}" "/etc/passwd" "/projects"; do \
+      echo "Changing permissions on ${f}" && chgrp -R 0 ${f} && \
+      chmod -R g+rwX ${f}; \
+    done && \
+    rustup component add rust-src; \
```
Comment (benoitf) [inline sidecars/rust/Dockerfile:11]: out of curiosity: how did we ended it up with these 3 components ?
what if they're not added at all

Comment (sunix) [reply sidecars/rust/Dockerfile:11]: These are required by the rust extension.
The rust extension is trying to install them at startup through provided dynamic task but fails in Che-theia (failed to run provided tasks, i will create an issue afterward, as it cannot run other provided tasks neither)

Comment (sunix) [reply sidecars/rust/Dockerfile:11]: these are required by the extension: https://github.com/rust-lang/vscode-rust/blob/master/src/rls.ts#L17
the extension is trying to install them if not present. but fails to run the tasks dynamically in Che-theia.
I will create an issue about that afterwards as some other provided tasks are failing to run.

**Review by benoitf — APPROVED** [review]

File: sidecars/rust/etc/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
```
Comment (ericwill) [inline sidecars/rust/etc/entrypoint.sh:1]: Is this file executable (i.e. `chmod +x` was run)?

Comment (sunix) [reply sidecars/rust/etc/entrypoint.sh:1]: yes it is executable
Selection_080

**Review by perenolde — APPROVED** [review]

**Pull request merged**